### ISSUE TITLE
Unable to Install Package in SandBox v4.7.1

### ISSUE THREAD

**Issue by RygelOrionis** [jongpie/NebulaLogger#308]
### Package Edition of Nebula Logger

Unlocked Package

### Package Version of Nebula Logger

v4.7.1

### New Bug Summary

Your request to install package "Nebula Logger Plugin Framework Overhaul" was unsuccessful. None of the data or setup information in your [salesforce.com](http://salesforce.com/) organization was affected.

If your install continues to fail, contact Salesforce CRM Support through your normal channels and provide the following information.

Organization: -----------------
User: -------------------------
Package: Nebula Logger (04t5Y0000015lgB)
Error Number: 1453525547-2553102 (232459185)
Problem:

1.  Method does not exist or incorrect signature: void getSObjectField() from the type FieldSetMember
RelatedLogEntriesController: Method does not exist or incorrect signature: void getSObjectField() from the type FieldSetMember

2.  Method does not exist or incorrect signature: void getSObjectField() from the type FieldSetMember
RelatedLogEntriesController: Method does not exist or incorrect signature: void getSObjectField() from the type FieldSetMember

3.  Method does not exist or incorrect signature: void getFieldPath() from the type FieldSetMember
RelatedLogEntriesController: Method does not exist or incorrect signature: void getFieldPath() from the type FieldSetMember

4.  Method does not exist or incorrect signature: void getLabel() from the type FieldSetMember
RelatedLogEntriesController: Method does not exist or incorrect signature: void getLabel() from the type FieldSetMember

5.  Dependent class is invalid and needs recompilation:
 Class RelatedLogEntriesController : Method does not exist or incorrect signature: void getSObjectField() from the type FieldSetMember
RelatedLogEntriesController_Tests: Dependent class is invalid and needs recompilation:
 Class RelatedLogEntriesController : Method does not exist or incorrect signature: void getSObjectField() from the type FieldSetMember

6. (LoggerCore) No classes found for RelatedLogEntriesController_Tests.
LoggerCore: No classes found for RelatedLogEntriesController_Tests.

7. (LoggerLogManagement) No classes found for RelatedLogEntriesController_Tests.
LoggerLogManagement: No classes found for RelatedLogEntriesController_Tests.

8. (relatedLogEntries) Unable to find Apex action class referenced as 'RelatedLogEntriesController'.
relatedLogEntries: Unable to find Apex action class referenced as 'RelatedLogEntriesController'.

9. (LoggerAdmin) In field: apexClass - no ApexClass named RelatedLogEntriesController found
LoggerAdmin: In field: apexClass - no ApexClass named RelatedLogEntriesController found

10. (LoggerEndUser) In field: apexClass - no ApexClass named RelatedLogEntriesController found
LoggerEndUser: In field: apexClass - no ApexClass named RelatedLogEntriesController found

11. (LoggerLogViewer) In field: apexClass - no ApexClass named RelatedLogEntriesController found
LoggerLogViewer: In field: apexClass - no ApexClass named RelatedLogEntriesController found

**Comment by RygelOrionis** [comment]
Problem present compiling all classes AND only Package classes

**Comment by jongpie** [comment]
Hi @RygelOrionis thanks for reporting this! It's definitely a strange error that I haven't seen before. Is there any chance that your sandbox has an Apex class called `FieldSetMember`? And are you upgrading, or installing the package for the first time in your sandbox?

**Comment by RygelOrionis** [comment]
Hi @jongpie , good catch. I indeed have.
I guess, i need some refactoring.
i'll let you know if i'm able to install afterwards.
Ty.

**Comment by RygelOrionis** [comment]
**Sandbox: Package Nebula Logger Install Successful**

**Comment by RygelOrionis** [comment]
Thank You for the update. Actually the refactoring was an easy fix on my end. Just a couple of references.
But i guess your fix should remove the problem from other users installing the package.

**Comment by jongpie** [comment]
Sounds good, it should just be a few lines of code to change in Nebula Logger to avoid this issue in the future, but glad to hear you were able to get it working in your sandbox. If you run into any other issues, just let me know!

**Comment by jongpie** [comment]
@RygelOrionis I was able to fix the issue for `FieldSetMember` by updating [a single line of code](https://github.com/jongpie/NebulaLogger/commit/10ea49ea9137100b4d1060944262585662c96ee1), but I found several other similar types of references to the classes `FieldSet`, `SObjectType`, `SObjectField` and `Test` that would have caused similar issues in any org with Apex classes using those names. I've [fixed those issues too](https://github.com/jongpie/NebulaLogger/commit/439d24f5a2ebe4e0be5c4aad40a6365232234ffd), and included some extra classes in Nebula Logger's pipeline to prevent these issues from happening again in the future.

Thanks again for opening this issue - it led to finding (and fixing!) some other similar problematic code.
